lib/matplotlib/tests/test_axes.py

```diff
@@ -1190,6 +1190,12 @@ def test_pcolorargs():
         ax.pcolormesh(x, y, Z[:-1, :-1], shading="gouraud")
     with pytest.raises(TypeError):
         ax.pcolormesh(X, Y, Z[:-1, :-1], shading="gouraud")
+    x[0] = np.NaN
+    with pytest.raises(TypeError):
```

Comment:
I see that TypeError is used in these earlier tests, so maybe you are using it for consistency.  I think that in both cases ValueError would be more consistent with general mpl practice, and the way I think about such things.   This is probably something where we need to state guidelines.  @tacaswell, your thoughts?  (I think that a long time ago I proposed we should define our own exception subclass of ValueError for use in argument validation, something like `MPL_ArgumentError`.  Or maybe that idea never got further than my head.)